airbyte-webapp/src/components/JobItem/components/Logs.tsx

```diff
@@ -57,4 +68,39 @@ const Logs: React.FC<LogsProps> = ({ logsArray }) => {
   );
 };
 
+/**
+ * Matching the log's line number by time makes the following assumptions:
+ * 1. The log's lines are already ordered by time
+ * 2. The timestamps used are in the same timezone
+ * 3. The error is closer to the end of the log file than the beginning
+ * 4. Lines are matched by the start of a line with "YYYY-MM-DD HH:mm:ss" format, like "2022-05-17 23:00:19 DEBUG I am a log message"
+ */
+const getMatchingLineNumbers = (matchTimestamp: number | undefined, lines: string[] | undefined) => {
+  if (!matchTimestamp || !lines || lines.length === 0) {
+    return [];
+  }
+  const flooredMatchTimestamp = Math.floor(matchTimestamp / 1000) * 1000;
+
+  const resolutionOffset = 1000; // the resolution of the timestamps is in seconds
+  const matchingLineNumbers: number[] = [];
+
+  let lineCounter = lines.length - 1;
+  while (lineCounter >= 0) {
+    const datetime = dayjs.utc(lines[lineCounter], dateTimeFormat, false)?.toDate()?.getTime();
```

Comment:
If there is not a match, this will return `NaN`, should never throw... Thanks, dayJs!